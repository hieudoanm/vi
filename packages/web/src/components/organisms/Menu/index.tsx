import Link from 'next/link';
import React from 'react';

export type MenuProps = {
  active: string;
  links: { href: string; text: string }[];
};

const Menu: React.FC<MenuProps> = ({ active = '', links = [] }) => {
  return (
    <div className="border rounded-lg overflow-hidden">
      <div className="grid grid-cols-1 md:grid-cols-2">
        {links.map((link: { href: string; text: string }, index: number) => {
          const { href, text } = link;
          const last = links.length - 1 === index;
          const border = last
            ? ''
            : 'border-b border-r-0 md:border-b-0 md:border-r';
          const activeClass =
            active === text ? 'bg-black text-white font-semibold' : '';
          return (
            <div key={`link-${href}`} className={`${border} ${activeClass}`}>
              <Link href={href} passHref>
                <div className="text-center py-2 uppercase cursor-pointer">
                  {text}
                </div>
              </Link>
            </div>
          );
        })}
      </div>
    </div>
  );
};

Menu.displayName = 'Menu';
Menu.defaultProps = {};

export default Menu;
