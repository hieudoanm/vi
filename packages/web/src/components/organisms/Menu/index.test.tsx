import { render } from '@testing-library/react';
import Menu from '.';

describe('Menu', () => {
  test('render default', () => {
    const wrapper = render(<Menu active="" links={[]} />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
