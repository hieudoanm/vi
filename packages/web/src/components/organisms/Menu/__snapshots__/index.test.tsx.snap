// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu render default 1`] = `
<div>
  <div
    class="border rounded-lg overflow-hidden"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-2"
    />
  </div>
</div>
`;
